src/parser_fill_cmd.c

```diff
@@ -78,33 +78,33 @@ t_err	ft_save_heredoc(t_tkn_list **lst, t_cmd *new)
  * @brief Open file and save fd as fd_out.
  *
  * If append bool is true, open file in append mode.
+ * Names of outfiles are saved in string array.
  * @param lst 		List of token.
  * @param new 		New cmd to be filled.
  * @param append	Bool to indicate if append mode is required.
  * @return t_err 	ERR_OPEN, SUCCESS
  */
 t_err	ft_save_outfile(t_tkn_list **lst, t_cmd *new, bool append)
 {
-	int			fd_out;
 	t_tkn_list	*tmp;
 
 	tmp = *lst;
 	tmp = tmp->next;
 	new->outfile = 1;
-	if (new->fd_out > -1)
-		if (close(new->fd_out) < 0)
-			return (ERR_CLOSE);
-	if (append)
-		fd_out = open(tmp->content, O_WRONLY | O_APPEND | O_CREAT, 0644);
-	else
-		fd_out = open(tmp->content, O_RDWR | O_TRUNC | O_CREAT, 0644);
-	if (fd_out == -1)
-		return (ERR_OPEN);
-	new->fd_out = fd_out;
+	new->outfiles[new->out_pos] = ft_strdup(tmp->content);
+	if (!new->outfiles[new->out_pos])
+		return (ERR_MALLOC);
 	if (append)
+	{
+		new->append_switches[new->out_pos] = true;
 		new->append = 1;
```

Comment:
Which - after some sleep - makes sense, since there can only be one outfile :D But a heredoc in the chain would break it I think
outfile, append, heredoc = append is true

Reply:
sleep no good

Reply:
true